lib/rspec/core/formatters/documentation_formatter.rb

```diff
@@ -23,23 +26,43 @@ def example_group_started(notification)
 
         def example_group_finished(_notification)
           @group_level -= 1 if @group_level > 0
+          flush_messages
         end
 
         def example_passed(passed)
           output.puts passed_output(passed.example)
+          flush_messages
         end
 
         def example_pending(pending)
           output.puts pending_output(pending.example,
                                      pending.example.execution_result.pending_message)
+          flush_messages
         end
 
         def example_failed(failure)
           output.puts failure_output(failure.example)
+          flush_messages
+        end
+
+        def message(notification)
+          if @group_level.zero?
```

Comment:
This worries me for behavioural reasons, instead, I think we should keep track of wether we're running in an example, buffer if so, or print immediately if not. It means keeping track of `example_start` as well but thats ok.

Reply:
Okay it is done.